llama_cpp/llama.py

```diff
@@ -406,32 +405,14 @@ def __init__(
             )
         )
 
-        self._lora_adapter: Optional[llama_cpp.llama_lora_adapter_p] = None
-
-        if self.lora_path:
-            self._lora_adapter = llama_cpp.llama_lora_adapter_init(
-                self._model.model,
-                self.lora_path.encode("utf-8"),
-            )
-            if self._lora_adapter is None:
-                raise RuntimeError(
-                    f"Failed to initialize LoRA adapter from lora path: {self.lora_path}"
-                )
-
-            def free_lora_adapter():
-                if self._lora_adapter is None:
-                    return
-                llama_cpp.llama_lora_adapter_free(self._lora_adapter)
-                self._lora_adapter = None
-
-            self._stack.callback(free_lora_adapter)
+        # Dict from LoRA path to wrapper
+        self._lora_adapters_paths: Dict[str, internals.LlamaLoraAdapter] = {}
```

Comment:
The `Llama` wrapper maintains a map to the low-level wrappers. We use the LoRA adapter path to key these objects. In theory we could track them using some sort of handle object, but this seemed OK for now.